src/cata_tiles.cpp

```diff
@@ -644,6 +644,198 @@ void cata_tiles::set_draw_scale( int scale )
     tile_ratioy = ( static_cast<float>( tile_height ) / static_cast<float>( fontheight ) );
 }
 
+std::optional<std::tuple<const tile_type *, std::string>> cata_tiles::tile_type_search(
+            const std::string &id, TILE_CATEGORY category,
+            const std::string &subcategory, int subtile, int rota
+        )
```

Comment:
might be a better idea to use structs?

```c++
struct tile_type {
	tile_type* tt;
	std::string found_id;
};
```

```suggestion
auto cata_tiles::tile_type_search(
            const std::string &id, TILE_CATEGORY category,
            const std::string &subcategory, int subtile, int rota
        ) -> std::optional<tile_type>
```

Reply:
Added struct: https://github.com/cataclysmbnteam/Cataclysm-BN/pull/5031/files#diff-567dea2474d5637f37f5066aa4e52c19f755ef11f674795c59511c01ef4a4cf8R349